google/cloud/storage/tests/curl_download_request_integration_test.cc

```diff
@@ -72,6 +74,65 @@ TEST(CurlDownloadRequestTest, SimpleStream) {
   EXPECT_EQ(kDownloadedLines, count);
 }
 
+// Run one attempt of the Regression7501 test. This is wrapped in a retry loop,
+// as integration tests flake due to unrelated (and unavoidable) problems, e.g.,
+// trying to setup connections.
+Status AttemptRegression7501() {
+  // Download the maximum number of lines supported by httpbin.org
+  auto constexpr kDownloadedLines = 100;
+  auto constexpr kTestPoolSize = 32;
+  auto factory =
+      std::make_shared<PooledCurlHandleFactory>(kTestPoolSize, Options{});
+
+  auto make_download = [factory] {
+    CurlRequestBuilder builder(
+        HttpBinEndpoint() + "/stream/" + std::to_string(kDownloadedLines),
+        factory);
+    return builder.BuildDownloadRequest(std::string{});
+  };
+
+  auto constexpr kBufferSize = kDownloadedLines;
+  char buffer[kBufferSize];
+
+  {
+    auto r_no_close = make_download();
+    auto read = r_no_close.Read(buffer, kBufferSize);
+    if (!read) return std::move(read).status();
+  }
+
+  {
+    auto r_partial_close = make_download();
+    auto read = r_partial_close.Read(buffer, kBufferSize);
+    if (!read) return std::move(read).status();
+    auto close = r_partial_close.Close();
+    if (!close) return std::move(close).status();
+  }
+
+  auto r_full = make_download();
```

Comment:
Do you need another no close read before the full read?

Reply:
No, I want to check that a handle where (a) not all the data is downloaded, and (b) you do call `Close()`, is usable.